src/main/java/org/datadog/jenkins/plugins/datadog/configuration/api/key/DatadogTextApiKey.java

```diff
@@ -30,6 +31,19 @@ public DatadogTextApiKey(Secret key) {
         this.key = key;
     }
 
+    /**
+     * Invoked by XStream when this object is deserialized.
+     * Ensures environment variables have higher priority than configuration persisted on disk
+     */
+    protected Object readResolve() {
+        Secret defaultKey = DatadogTextApiKeyDescriptor.getDefaultKey();
+        if (defaultKey != null){
+            return new DatadogTextApiKey(defaultKey);
+        } else {
+            return this;
```

Comment:
So basically, `this` in this context is the instance deserialized using the `xml` config file?
Could we add a comment on this line indicating this?

Reply:
> this in this context is the instance deserialized using the xml config file
Correct. Added the comment explaining it